BLSQ/bluesquare-components
# IA-1897: refactor SearchInput component

`SearchFilter` in iaso  was only used in OrgUnits, i removed the component and refactored `SearchInput`  to handle text search errors and use `InpuComponent` in `OrgunitsFilter` 

Related JIRA tickets : [IA-1897](https://bluesquare.atlassian.net/browse/IA-1897?atlOrigin=eyJpIjoiYzllMWYyZDI4NWFmNDBhNzkyM2ZkODU5YTU2YWQ0OWQiLCJwIjoiaiJ9)

## Self proofreading checklist


## Changes

- use TypeScript
- add `onErrorChange` prop to handle enable/disable Search button
- add `blockForbiddenChars` prop to allow component to do conditional check on special characters
- add `textSearchErrors` state to pass error message to display

## How to test

Explain how to test your PR.
See the related PR on iaso for testing: [IA-1897: replace SearchFilter component](https://github.com/BLSQ/iaso/pull/414)

[IA-1897]: https://bluesquare.atlassian.net/browse/IA-1897?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
